### ISSUE TITLE
‘tf.train.shuffle_batch’ problem

### ISSUE THREAD

**Issue by friedhelm739** [tensorflow/tensorflow#21033]
When I train my face_detection model，I set 0 as my non-face label ,and 1 as face label .After building TFRecord dataset，I use 'image_batch, label_batch = tf.train.shuffle_batch ' ,it  is clear  that ‘image_batch ’ corresponds to ‘label_batch’   .However,I found that ‘image_batch ’ is not corresponding to ‘label_batch’. some non-face pictures'label is 1,and some face pictures' label is 0,why?
here is my code:`import tensorflow as tf
import time
import matplotlib.pyplot as plt

begin=time.time()
filename_queue = tf.train.string_input_producer(['C:\\Users\\312\\Desktop\\Face_Detection\\ai.tfrecords'],shuffle=False)

reader = tf.TFRecordReader()
_, serialized_example = reader.read(filename_queue)

features = tf.parse_single_example(serialized_example,
                               features={
                               'label':tf.FixedLenFeature([],tf.int64),
                               'img_raw':tf.FixedLenFeature([],tf.string),
                               })
img=tf.decode_raw(features['img_raw'],tf.uint8)
img = tf.reshape(img, [227,227,3])
label=tf.cast(features['label'],tf.int32)

min_after_dequeue = 200
batch_size = 128
capacity = min_after_dequeue + 10 * batch_size
image_batch, label_batch = tf.train.batch([img, label], batch_size=batch_size, capacity=capacity)

image_batch, label_batch = tf.train.shuffle_batch(
      [img, label],
      batch_size=32,
      num_threads=7,
      capacity=500,
      min_after_dequeue=200)

with tf.Session() as sess:
    sess.run((tf.global_variables_initializer(),
              tf.local_variables_initializer()))
    coord = tf.train.Coordinator()
    threads = tf.train.start_queue_runners(sess=sess,coord=coord)
    for i in range(1): 
        label=sess.run(label)
        label_batch=sess.run(label_batch)
        image_batch=sess.run(image_batch) 

        for j in range(8):
            plt.imshow(image_batch[j])
            print(label_batch[j])
            plt.show()
    coord.request_stop()
    coord.join(threads)
print(time.time()-begin)
    `

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by skye** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
